src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -253,9 +256,8 @@ public void validate() throws AppEngineException {
               + getDevAppServerPath() + "' is not a file.");
     }
     if (!Files.isDirectory(getJavaAppEngineSdkPath())) {
-      throw new AppEngineException(
-          "Validation Error: Java App Engine SDK location '"
-              + getJavaAppEngineSdkPath() + "' is not a directory.");
+      throw new AppEngineJavaComponentsNotInstalledException(
+          "Validation Error: Java App Engine components not installed");
```

Comment:
What do you think about adding information to the exception message on how to install it (e.g. 'run gcloud components install ...'). Or do you think its better left to the clients to do this if needed?